### ISSUE TITLE
Mail is send as somebody when using Circles

### ISSUE THREAD

**Issue by Gerben-W** [nextcloud/circles#1447]
When sharing files with a single mail address, the subject of the mail contains the user that shares this file. 
( USERNAME shared »FILENAME« with you )

But when sharing to a Circle, the subject of the mail contains "someone".
( someone shared a file with you. )

Is it possible to change the mail subject so that the username is in the subject just like sharing a file with a single mail address?

**Comment by Noki77** [comment]
I diagnosed this a fair bit now, since i saw this as well after a sharing attempt of one of our customers bounced...
seems like the issue is caused by `ShareWrapper::hasInitiator` returning `false` in [ShareCreatedSendMail.php#L165](https://github.com/nextcloud/circles/blob/master/lib/Listeners/Files/ShareCreatedSendMail.php#L165), though I haven't figured out why yet...
In my case I "fixed" the issue by simply replacing "someone" within the mail subject by using an event listener on the  `BeforeMessageSent` Event in a custom NextCloud App, that i'm using anyway for having certain email templates based on the user mail address or groups.

So if you still want to "fix" this yourself and you're willing to go through the effort of making an own NC app for it, here is basically what you need to do:
* Create a custom app (just follow the [docs](https://docs.nextcloud.com/server/latest/developer_manual//getting_started/index.html) or use the [app skeleton generator by NextCloud](https://apps.nextcloud.com/developer/apps/generate))
* write an event listener on `\OCA\Circles\Events\AddingCircleMemberEvent`, that fetches the circle owner and stores it somewhere (in my case I have an `Application::setDispatchingUser` function
* write an event listener on `OCP\Mail\Events\BeforeMessageSent` that first checks whether a circle owner was found earlier (in my case just check whether `Application::getDispatchingUser()` is not null), then whether `$event->getMessage()->getSubject()` contains  the word "someone" (yes, IMessage doesn't have a `getSubject()` function, but `\OC\Mail\Message` does, maybe check if it's an instance of it first) and lastly replace "someone" with the previously fetched circle owners' display name
* register both listeners in your `Application::register` method

**Comment by Gerben-W** [comment]
@Noki77 Thanks for helping with this.
Loved your solution and thinking of implementing this.
Unfortunately I only have one concern you might be able to help with.
In the mail which is send out, the body contains the Circle name, which can be anything (even non positive names to group a certain amount of people). Do you know how to easily replace the circle name with the word "you" in the mail that is send out.
Basically that the mail looks more like the standard mail that is send to one person in stead of a Circle.

Thank you.

**Comment by Noki77** [comment]
@Gerben-W happy to hear it works out for you! 
Good idea with the circle name, i'll look into how it could be implemented.

**Comment by spanje** [comment]
@Noki77 : It would be really great if you could implement this :-)

**Comment by Jerome-Herbinet** [comment]
I'm still having that problem with the last contacts + teams apps in NC 29. Mail is sent by "someone" (in e-mails subject an body). That's pretty weird and may damage UX. @ArtificialOwl or @nickvergessen, what do you think ?

Here is what I get (IMO, the share issuer should be indicated instead of "someone", because without that information, some recipients could suspect spamming) : 

![2024-08-21_14-29](https://github.com/user-attachments/assets/49d36f5d-8791-4eaa-99a6-2e08a1991a9f)
![2024-08-21_14-30](https://github.com/user-attachments/assets/8153d2d5-0361-4188-9d38-b0a1cabb555f)

**Comment by Jerome-Herbinet** [comment]
> I can point you to the search results:
> 
> https://github.com/nextcloud/circles/blob/f725ecaffbfae29a98ef0eea745d5bcf6e64b90c/lib/Listeners/Files/ShareCreatedSendMail.php#L131
> 
> https://github.com/nextcloud/circles/blob/f725ecaffbfae29a98ef0eea745d5bcf6e64b90c/lib/Listeners/Files/MemberAddedSendMail.php#L95
> 
> String is not translated and even if translated it should not be inserted into a string like that as it might not work in every language.
> 
> But I'm not the maintainer of the app, so I'm unsubscribing again

How can I get the maintainers' list of an app, in order to ping the right persons ?
